### ISSUE TITLE
Keeping API backward compatible: PluginConfiguration#getActiveLocation

### ISSUE THREAD

**Issue by vicky17d** [jshiell/checkstyle-idea#567]
org.infernus.idea.checkstyle.config.PluginConfiguration

```
    public ConfigurationLocation getActiveLocation() {
        return activeLocation;
    }
```
has been changed to 

```
   public SortedSet<ConfigurationLocation> getActiveLocations() {
      return this.activeLocations;
   }
```

This change breaks plugins extending the behavior of checkstyle-idea plugin.
Since IntelliJ shows _**Update**_ available on plugins as soon as the new version is available, there is no way to automatically stop users from upgrading and finding a bunch of exceptions.
Any chance you can look into keeping the old method (even for a short time and mark it deprecated) and return the getActiveLocations.first()

**Comment by jshiell** [comment]
I'll have a think, but looks unlikely at present, as the fix for #565 means changing the new method to `public SortedSet<ConfigurationLocation> getActiveLocations(Project project)`.

**Comment by jshiell** [comment]
Sorry, this will be a no, as the fix for #565 does need to resolve the active descriptor, which means calling into IDEA's silly DI framework that wants JIT lookups. And I'm afraid I'm not willing to do the hacks require to keep a ref to the project in the object purely for this case.
